crates/editor/src/editor.rs

```diff
@@ -3328,6 +3328,33 @@ impl Editor {
         Subscription::join(other_subscription, this_subscription)
     }
 
+    fn unfold_buffers_with_selections(&mut self, cx: &mut Context<Self>) {
+        if self.buffer().read(cx).is_singleton() {
+            return;
+        }
+
+        let snapshot = self.buffer.read(cx).snapshot(cx);
+        let buffers_to_unfold: Vec<BufferId> = self
+            .selections
+            .disjoint_anchor_ranges()
+            .flat_map(|range| {
+                // Check all buffers that this selection spans
+                let start_offset = range.start.to_offset(&snapshot);
+                let end_offset = range.end.to_offset(&snapshot);
+                snapshot.buffer_ids_for_range(start_offset..end_offset)
+            })
+            .filter(|buffer_id| self.is_buffer_folded(*buffer_id, cx))
```

Comment:
I think instead of filtering here, which may lead to multiple duplicate `is_buffer_folded` calls for the same buffer, it might be better to perform this step after creating the HashSet.